sessions/React-Typescript-100/101/101.003/README.md

```diff
@@ -1 +1,182 @@
-Adding Components and passing props
\ No newline at end of file
+# React example without JSX
+Setting up project without create-react-app (using EsBuild):
+
+  1. Use terminal: `git init`
+ 
+  2. Create folders: 
+     - `src`: used for our application source code
+     - `public`: used for our `.html` file
+  
+  3. Add `build`,`temp`, `node_modules` folders to `.gitignore` file
+  4. Initialise the npm project/package
+     - Use terminal: `npm init`
+
+  5. Create file `/public/index.html`
+
+         <!DOCTYPE html>
+         <html lang="en">
+           <head>
+               <meta charset="UTF-8">
+               <meta name="viewport" content="width=device-width, initial-scale=1.0">
+               <meta http-equiv="X-UA-Compatible" content="ie=edge">
+               <title>Document</title>
+           </head>
+           <body>
+               <div id="app"></div>
+               <script src="bundle.js"></script>
+           </body>
+         </html>
+
+  6. Add essential dependencies for ReactJs
+     - Use terminal: `yarn add react react-dom` 
+
+  7. To use TypeScript with our React project, we need some additional dev dependencies
+     - Use terminal: `yarn add -D typescript @types/react @types/react-dom` 
+
+  8. Create new file: `tsconfig.json` (TODO: Look at the one created in typescript 101 sessions)
+     
+         {
+           "include": [
+               "src/*",
+               "./declarations.d.ts"
+           ],
+           "compilerOptions": {
+               "target": "es5",
+               "jsx": "react",
+               "allowSyntheticDefaultImports": true
+           }
+         }
+
+  9. Create file: `src/App.tsx`
+
+         import React from 'react';
+      
+         const App = () => (
+             <h1>React App</h1>
+         );
+      
+         export default App;
+
+  10. Create new file: `src/index.tsx`
+
+          import React from 'react';
+          import ReactDOM from 'react-dom';
+          import App from './App';
+     
+          ReactDOM.render(<App />, document.getElementById('app'));
+  
+  11. EsBuild is the module bundler which will compile all the javascript files or modules into single file called as bundle.
+         - Use terminal: `yarn add esbuild esbuild-plugin-inline-image esbuild-css-modules-plugin`
+         - Use terminal: `yarn add -D live-server`
+         - Create file `serve.js`
+
+               const cssModulesPlugin = require("esbuild-css-modules-plugin");
+               const inlineImage = require("esbuild-plugin-inline-image");
+               const {build} = require("esbuild")
+               const liveServer = require("live-server")
+    
+               ;(async () => {
```

Comment:
Is the ; correct here

Reply:
Yeah, it's is commonly used in JS and is borrowed from early days of JS. A good article to read if you like: https://medium.com/@maxbendick/semicolons-go-at-the-beginning-of-the-line-34ea9d26393a

Reply:
Changed the style of script